Make passing a file on command line consistent with :load with regards to loading configuration

Strahinja opened the pull request:
This PR rearranges the order of `load_rc` and `load_file` and also removes some related redundant checks.

Related issue: #778

andmarti1424 commented:
@Strahinja thank you for your work. could you please explain a little more what this PR achieves/solves? Thanks!

Strahinja commented:
> @Strahinja thank you for your work. could you please explain a little more what this PR achieves/solves? Thanks!

 As [stated][1] in #778, and per [your request][2], I made this PR to fix the issue that the variable (now renamed to `import_delimited_to_text`) has no effect when a file is loaded by giving the pathname to sc-im on the command line. (Using the command `:load` to load the file from within sc-im respects the variable.) The override:
```
sc-im --import_delimited_to_text myfile.tsv
```
works, but simply passing the pathname without the `--import_delimited_to_text` switch, while having
```
set import_delimited_to_text=1
```
in the configuration file, doesn't work. Since the order of `load_file` and `load_rc` is swapped, the check
```
if (session->cur_doc == NULL) create_empty_wb();
```
is redundant, so I also removed it. The function `create_empty_wb` is always called.

[1]: https://github.com/andmarti1424/sc-im/issues/778#issuecomment-1416182629
[2]: https://github.com/andmarti1424/sc-im/issues/778#issuecomment-1434511951

andmarti1424 commented:
@Strahinja yes! sorry. my memory is so bad. I remember now. Will merge this today.

andmarti1424 commented:
@Strahinja I have noticed

andmarti1424 commented:
@Strahinja I have noticed if you open a file ./sc-im ../examples/sc/sheets/a.sc
it create a new wb even though its not needed. please see screenshot below.

Strahinja commented:
My apologies, I haven't noticed that. I still think the logical solution should be to move the configuration loading code before the loading of the file, since there are configuration variables affecting the loading. The part regarding the NULL check can be reverted, but IMHO the configuration loading code should be made independent of the file loading code.

andmarti1424 commented:
The move you made is ok. it should be kept as it is now. the only thing to be done is check for the null..

andmarti1424 commented:
Please update to latest commit on dev branch and check it out.

Strahinja commented:
With my fork rebased to dev branch and with some of the changes reverted, the setting doesn't seem to be obeyed at all, both with specifying pathnames and `:load`. My `$HOME/.config/sc-im/scimrc`:
```
set import_delimited_to_text=1
```

andmarti1424 commented:
@Strahinja this was ok:
![image](https://github.com/andmarti1424/sc-im/assets/9466161/6da8e0d5-8ef9-44e0-bb1f-4971d1166bff)
Why did you revert it?
I have also changed read_file function in file.c
Please check you have the same version as current dev branch in here: https://github.com/andmarti1424/sc-im/commit/b1ba526d0be4b4ceff13b4d89ebfcaba954d26a3

Strahinja commented:
Ok, I pulled the latest dev, rebased my fork and reintroduced the changes I made again, meaning the code in `src/main.c` reads:

https://github.com/Strahinja/sc-im/blob/de25a92b8c107a1629fda1dfec920fdc4e02d60a/src/main.c#L270-L284

Also, I have:

https://github.com/Strahinja/sc-im/blob/de25a92b8c107a1629fda1dfec920fdc4e02d60a/src/file.c#L2215-L2226

but after `make -C src clean all` it seems that the configuration setting/variable `import_delimited_to_text` is still not obeyed, meaning I got this:
![2023-06-02-152359-dwm](https://github.com/andmarti1424/sc-im/assets/981077/b8a07aad-9ba0-44cc-a251-0410d96ac8dc)

andmarti1424 commented:
Yes. Something seems wrong. :(

andmarti1424 commented:
I got this. Found the problem.

andmarti1424 commented:
The configuration variable name was wrong in file.c. Please update to latest commit on dev branch, or just replace import_delimited_as_text to import_delimited_to_text in file.c

Strahinja commented:
Yes, I can confirm that works!
